src/OpenConext/ProfileBundle/Service/ConsentService.php

```diff
@@ -57,34 +42,6 @@ public function __construct(
      */
     public function findAllFor(AuthenticatedUser $user)
     {
-        if (!$user->getAttributes()->containsAttributeDefinedBy($this->identifyingAttribute)) {
-            $message = sprintf(
-                'Cannot get consent list for user: user does not have identifying attribute "%s"',
-                $this->identifyingAttribute->getName()
-            );
-
-            $this->logger->error($message);
-
-            return null;
-        }
-
-        $userIdentifier    = $user->getAttributes()->getAttributeByDefinition($this->identifyingAttribute);
-        $identifyingValues = $userIdentifier->getValue();
-        $identifyingValue  = array_shift($identifyingValues);
-
-        if (!is_string($identifyingValue)) {
-            $message = sprintf(
-                'In order to get the consent list for a user, the identifying attribute must have a string value. "%s"'
-                . 'given for identifying attribute "%s"',
-                gettype($identifyingValue),
-                $this->identifyingAttribute->getName()
-            );
-
-            $this->logger->error($message);
-
-            return null;
-        }
-
-        return $this->consentRepository->findAllFor($identifyingValue);
+        return $this->consentRepository->findAllFor($user->getNameId());
```

Comment:
Shouldn't there be any validation on the nameId? Because the AssertionAdapter could return null as value and maybe we want to log that.

Reply:
`\OpenConext\Profile\Entity\AuthenticatedUser::__construct` asserts that the NameId is a string. There could be a very rare case that the NameID value is an empty string. Then, the `\OpenConext\Profile\Repository\ConsentRepository::findAllFor` method could be called with an empty string. But I don't see that happening.